includes/HttpFunctions.php

```diff
@@ -82,11 +82,18 @@ public static function request( $method, $url, $options = array() ) {
 				'isOk' => $isOk,
 				'requestTimeMS' => $requestTime,
 				'backendTimeMS' => intval( 1000 * $backendTime),
+				'responseHeaders' => $req->getResponseHeaders(),
 			];
 			if ( !$isOk ) {
-				$params[ 'statusMessage' ] = $status->getMessage();
+				$params[ 'statusMessage' ] = $status;
+				$params[ 'exception' ] = new WikiaException( "{$caller} - HTTP request failed", $req->getStatus() );
+				$level = 'error';
```

Comment:
@nandy-andy: this will allow us to automatically report flood of failed HTTP requests (grouped by the method that calls `Http::get` / `Http::post` helper).